In [31]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

import warnings 
warnings.filterwarnings("ignore")

from IPython.display import display

In [59]:
file_path_1 = 'D:/MSBA7024/復本 148_56.xlsx'
Non_sale_Peak = pd.read_excel(file_path_1, sheet_name='商品明細報表') 

file_path_2 = 'D:/MSBA7024/複本 148_1011.xlsx'
Non_sale_Non_peak = pd.read_excel(file_path_2, sheet_name='商品明細報表') 

file_path_3 = 'D:/MSBA7024/複本 1911_56.xlsx'
Sale_Peak = pd.read_excel(file_path_3, sheet_name='商品明細報表') 

file_path_4 = 'D:/MSBA7024/複本 1911_1011.xlsx'
Sale_Non_peak = pd.read_excel(file_path_4, sheet_name='商品明細報表') 

In [60]:
combined_df = pd.concat([Non_sale_Peak, Non_sale_Non_peak, Sale_Peak, Sale_Non_peak], axis=0)

In [61]:
Non_sale_Non_peak

Sample Transaction Data - 2023 Aug  ( Non Salesday - 1小時 - 非繁忙時段)  \
0                                                  門店編碼                  
1                                                   001                  
2                                                   001                  
3                                                   001                  
4                                                   001                  
...                                                 ...                  
7131                                                012                  
7132                                                012                  
7133                                                012                  
7134                                                012                  
7135                                                012                  

          Unnamed: 1      Unnamed: 2 Unnamed: 3        Unnamed: 4  \
0           銷售單號 (N)    DEPARTMENT名稱    SECTION         SECTION名稱   
1     ****5443369702   FOOD & DELICA        131           GROCERY   
2     ****5443377218   FOOD & DELICA        136              FISH   
3     ****5443381605   FOOD & DELICA        134             DAILY   
4     ****5443381605   FOOD & DELICA        134             DAILY   
...              ...             ...        ...               ...   
7131  ****5473941228    MISC-PACKING        401     SHOPPING BAGS   
7132  ****5473941228   FOOD & DELICA        134             DAILY   
7133  ****5474001321  HOUSEHOLD & EA        171          NON FOOD   
7134  ****5474005225  HOUSEHOLD & EA        165  LIVING APPLIANCE   
7135  ****5474034016   FOOD & DELICA        134             DAILY   

                 Unnamed: 5  Unnamed: 6 Unnamed: 7 Unnamed: 8      Unnamed: 9  
0                      商品名稱      貨號 (N)        銷售價         數量        會員ID (N)  
1                     維他蒸餾水  ***2589851        3.5          1             NaN  
2                       飛魚子  ***0948356       31.7          1             NaN  
3                 澳洲保利 脫脂鮮奶  ***2473429       42.5          1  *******1751342  
4                墨爾本 三文治芝士片  ***2744159       29.9          1  *******1751342  
...                     ...         ...        ...        ...             ...  
7131             超市膠袋 (加大碼)  ***0000051          1          1             NaN  
7132                   蟹柳薄片  ***2898377       42.9          1             NaN  
7133   TV 成人紙尿褲薄裝 中至大碼 長時間用  ***0330229        129          1             NaN  
7134  MASTERPLUG 轉換插頭 SHADC  ***0532196         15          1             NaN  
7135         OURHOME 韓國蒸煮豆腐  ***0367124       13.9          1             NaN  

[7136 rows x 10 columns]

## EDA

In [62]:
# Rename column
Non_sale_Peak.columns = ['store_code', 'sales_order', 'department', 'section_code', 'section_name', 'product_name', 'product_code', 'sales_price', 'quantity', 'member_id']
Non_sale_Non_peak.columns = ['store_code', 'sales_order', 'department', 'section_code', 'section_name', 'product_name', 'product_code', 'sales_price', 'quantity', 'member_id']
Sale_Peak.columns = ['store_code', 'sales_order', 'department', 'section_code', 'section_name', 'product_name', 'product_code', 'sales_price', 'quantity', 'member_id']
Sale_Non_peak.columns = ['store_code', 'sales_order', 'department', 'section_code', 'section_name', 'product_name', 'product_code', 'sales_price', 'quantity', 'member_id']

# Transfer price and quantity to numeric
Non_sale_Peak['sales_price'] = pd.to_numeric(Non_sale_Peak['sales_price'], errors='coerce')
Non_sale_Peak['quantity'] = pd.to_numeric(Non_sale_Peak['quantity'], errors='coerce')
Non_sale_Non_peak['sales_price'] = pd.to_numeric(Non_sale_Non_peak['sales_price'], errors='coerce')
Non_sale_Non_peak['quantity'] = pd.to_numeric(Non_sale_Non_peak['quantity'], errors='coerce')
Sale_Peak['sales_price'] = pd.to_numeric(Sale_Peak['sales_price'], errors='coerce')
Sale_Peak['quantity'] = pd.to_numeric(Sale_Peak['quantity'], errors='coerce')
Sale_Non_peak['sales_price'] = pd.to_numeric(Sale_Non_peak['sales_price'], errors='coerce')
Sale_Non_peak['quantity'] = pd.to_numeric(Sale_Non_peak['quantity'], errors='coerce')

# Handle the member id
Non_sale_Peak['member_id'] = Non_sale_Peak['member_id'].fillna('nonmember')
Non_sale_Non_peak['member_id'] = Non_sale_Non_peak['member_id'].fillna('nonmember')
Sale_Peak['member_id'] = Sale_Peak['member_id'].fillna('nonmember')
Sale_Non_peak['member_id'] = Sale_Non_peak['member_id'].fillna('nonmember')

# Delete the first row
Non_sale_Peak = Non_sale_Peak.iloc[1:].reset_index(drop=True)
Non_sale_Non_peak = Non_sale_Non_peak.iloc[1:].reset_index(drop=True)
Sale_Peak = Sale_Peak.iloc[1:].reset_index(drop=True)
Sale_Non_peak = Sale_Non_peak.iloc[1:].reset_index(drop=True)

In [63]:
display(Non_sale_Peak, Non_sale_Non_peak, Sale_Peak, Sale_Non_peak)

store_code     sales_order        department section_code  section_name  \
0            001  ****5487981015      LIVING PLAZA          709  LIVING PLAZA   
1            001  ****5487981015      LIVING PLAZA          709  LIVING PLAZA   
2            001  ****5487989030  MISC-OTHER SALES          864         OTHER   
3            001  ****5488003511    HOUSEHOLD & EA          164  HOUSEKEEPING   
4            001  ****5488011125           FASHION          117          MENS   
...          ...             ...               ...          ...           ...   
19637        012  ****5530161026     FOOD & DELICA          134         DAILY   
19638        012  ****5530207506      LIVING PLAZA          709  LIVING PLAZA   
19639        012  ****5530207506      LIVING PLAZA          709  LIVING PLAZA   
19640        012  ****5530207506      LIVING PLAZA          709  LIVING PLAZA   
19641        012  ****5530207506      LIVING PLAZA          709  LIVING PLAZA   

                             product_name product_code  sales_price  quantity  \
0      TRANSPARENT CUP (SWEET TIME 300ML)  ***05740873         12.0       1.0   
1                 GENTLE-WASH BODY SPONGE  ***05530811         12.0       1.0   
2                         HSBC COUPON $50  ***06003735         50.0      15.0   
3                      TVHC 速乾面巾 黃 (4/96)  ***30469324         59.9       1.0   
4                      CONSIG.- S & F INT  ***09927005        238.0       1.0   
...                                   ...          ...          ...       ...   
19637                        日本之惠 白蛋 10隻裝  ***20284626         28.9       2.0   
19638      PLASTIC HANGING POT MINI WHITE  ***50292051         12.0       1.0   
19639             GLASS VASE(PEAR-SHAPED)  ***50601376         12.0       2.0   
19640          BOWL -HEXAGON-4.52IN-BROWN  ***50597970         12.0       1.0   
19641             GLASS VASE(CLEAR LABEL)  ***50600246         12.0       2.0   

            member_id  
0           nonmember  
1           nonmember  
2           nonmember  
3           nonmember  
4           nonmember  
...               ...  
19637  *******2022864  
19638       nonmember  
19639       nonmember  
19640       nonmember  
19641       nonmember  

[19642 rows x 10 columns]

store_code     sales_order      department section_code  \
0           001  ****5443369702   FOOD & DELICA          131   
1           001  ****5443377218   FOOD & DELICA          136   
2           001  ****5443381605   FOOD & DELICA          134   
3           001  ****5443381605   FOOD & DELICA          134   
4           001  ****5443381605   FOOD & DELICA          131   
...         ...             ...             ...          ...   
7130        012  ****5473941228    MISC-PACKING          401   
7131        012  ****5473941228   FOOD & DELICA          134   
7132        012  ****5474001321  HOUSEHOLD & EA          171   
7133        012  ****5474005225  HOUSEHOLD & EA          165   
7134        012  ****5474034016   FOOD & DELICA          134   

          section_name           product_name product_code  sales_price  \
0              GROCERY                  維他蒸餾水   ***2589851          3.5   
1                 FISH                    飛魚子   ***0948356         31.7   
2                DAILY              澳洲保利 脫脂鮮奶   ***2473429         42.5   
3                DAILY             墨爾本 三文治芝士片   ***2744159         29.9   
4              GROCERY         UCC 即溶咖啡NO.114   ***2051563         49.9   
...                ...                    ...          ...          ...   
7130     SHOPPING BAGS             超市膠袋 (加大碼)   ***0000051          1.0   
7131             DAILY                   蟹柳薄片   ***2898377         42.9   
7132          NON FOOD   TV 成人紙尿褲薄裝 中至大碼 長時間用   ***0330229        129.0   
7133  LIVING APPLIANCE  MASTERPLUG 轉換插頭 SHADC   ***0532196         15.0   
7134             DAILY         OURHOME 韓國蒸煮豆腐   ***0367124         13.9   

      quantity       member_id  
0          1.0       nonmember  
1          1.0       nonmember  
2          1.0  *******1751342  
3          1.0  *******1751342  
4          1.0  *******1751342  
...        ...             ...  
7130       1.0       nonmember  
7131       1.0       nonmember  
7132       1.0       nonmember  
7133       1.0       nonmember  
7134       1.0       nonmember  

[7135 rows x 10 columns]

store_code     sales_order      department section_code  \
0            001  ****9038854327  HOUSEHOLD & EA          165   
1            001  ****9038864013  HOUSEHOLD & EA          165   
2            001  ****9038867517         FASHION          118   
3            001  ****9038867517         FASHION          118   
4            001  ****9038880116   FOOD & DELICA          133   
...          ...             ...             ...          ...   
45774        012  ****9102047508   FOOD & DELICA          131   
45775        012  ****9102118222   FOOD & DELICA          135   
45776        012  ****9102129521   FOOD & DELICA          131   
45777        012  ****9102129521   FOOD & DELICA          131   
45778        012  ****9123643723   FOOD & DELICA          135   

           section_name                       product_name product_code  \
0      LIVING APPLIANCE               TURBO 男士鼻毛剪套裝 THT200  ***30522908   
1      LIVING APPLIANCE                       伊瑪子母燉盅 ISP16  ***00769141   
2             UNDERWEAR   TV158SA5303H PFWM綿 男裝V領長袖內衣 BK L  ***12490298   
3             UNDERWEAR  TV158SA5403H PFWM綿 男裝圓領長袖內衣 CGY L  ***12490488   
4                DELICA                           炸蒜照燒雞扒便當  ***00405852   
...                 ...                                ...          ...   
45774           GROCERY                      森永 迪士尼朱古力味夾心餅  ***02528008   
45775           PRODUCE                            日本紅?甜蕃薯  ***20396396   
45776           GROCERY                           李錦記煲仔飯醬油  ***02392959   
45777           GROCERY                         日本 故里風味蕎麥麵  ***02505618   
45778           PRODUCE                          紐西蘭珍寶金奇異果  ***20242566   

       sales_price  quantity       member_id  
0            208.0       1.0  *******2027995  
1            290.0       1.0       nonmember  
2             79.0       1.0  *******1758861  
3             79.0       1.0  *******1758861  
4             39.9       1.0       nonmember  
...            ...       ...             ...  
45774          9.9       1.0  *******1812293  
45775         24.9       1.0  *******1880242  
45776          9.9       1.0       nonmember  
45777         15.9       2.0       nonmember  
45778         26.9       1.0       nonmember  

[45779 rows x 10 columns]

store_code     sales_order      department section_code section_name  \
0            001  ****8955107426  BENTO & BAKERY          138       BAKERY   
1            001  ****8955107426  BENTO & BAKERY          138       BAKERY   
2            001  ****8955107426  BENTO & BAKERY          138       BAKERY   
3            001  ****8955107426  BENTO & BAKERY          138       BAKERY   
4            001  ****8955116719  HOUSEHOLD & EA          172     COSMETIC   
...          ...             ...             ...          ...          ...   
15059        012  ****8985588706   FOOD & DELICA          131      GROCERY   
15060        012  ****8985658528   FOOD & DELICA          134        DAILY   
15061        012  ****8985658528   FOOD & DELICA          131      GROCERY   
15062        012  ****8985720217  HOUSEHOLD & EA          171     NON FOOD   
15063        012  ****8985723522   FOOD & DELICA          131      GROCERY   

                   product_name product_code  sales_price  quantity  member_id  
0                      迷你朱古力丹麥酥  ***00621383          4.0       1.0  nonmember  
1                 雀巢罐裝香濃咖啡250亳升  ***60003712          9.2       1.0  nonmember  
2                      迷你朱古力丹麥酥  ***00621383          4.0       1.0  nonmember  
3                      迷你朱古力丹麥酥  ***00621383          4.0       1.0  nonmember  
4                梨奧美 蔗糖白橙花滋潤沐浴露  ***00994244         35.9       1.0  nonmember  
...                         ...          ...          ...       ...        ...  
15059                  福字 米粉5包裝  ***02900124         22.9       1.0  nonmember  
15060                   榮鈦行 糯米雞  ***02036077         38.9       1.0  nonmember  
15061      SURF BEVERAGE 香印提子綠茶  ***20486981          9.9       1.0  nonmember  
15062  維達4D DELUXE袋裝面紙5包裝(天然無香)  ***30270300         27.5       1.0  nonmember  
15063        日本味の素 家樂牌 完熟番茄湯螺絲粉  ***20167201         16.9       4.0  nonmember  

[15064 rows x 10 columns]

## Overall analysis

In [75]:
data_frames = [Non_sale_Peak, Non_sale_Non_peak, Sale_Peak, Sale_Non_peak]

# ratio of member and nonmember
for df, name in zip(data_frames, ['Non_sale_Peak', 'Non_sale_Non_peak', 'Sale_Peak', 'Sale_Non_peak']):
    member_transactions = df[df['member_id'] != 'nonmember']
    non_member_transactions = df[df['member_id'] == 'nonmember']
    print(f"{name}:")
    print(f"Ratio of member transaction: {len(member_transactions) / len(df) * 100:.2f}%")
    print(f"Ratio of nonmember transaction: {len(non_member_transactions) / len(df) * 100:.2f}%")
    print()

Non_sale_Peak:
Ratio of member transaction: 31.09%
Ratio of nonmember transaction: 68.91%

Non_sale_Non_peak:
Ratio of member transaction: 37.18%
Ratio of nonmember transaction: 62.82%

Sale_Peak:
Ratio of member transaction: 50.38%
Ratio of nonmember transaction: 49.62%

Sale_Non_peak:
Ratio of member transaction: 2.56%
Ratio of nonmember transaction: 97.44%



In [76]:
# PB and NonPB
for df, name in zip(data_frames, ['Non_sale_Peak', 'Non_sale_Non_peak', 'Sale_Peak', 'Sale_Non_peak']):
    # PB items
    pb_items = df[df['product_name'].str.contains('TV|TVHC', case=False, na=False)]
    non_pb_items = df[~df['product_name'].str.contains('TV|TVHC', case=False, na=False)]
    
    total_items = len(df)
    total_sales_amount = df['sales_price'].sum()

    print(f"{name}:")
    # PB items
    pb_item_count = len(pb_items)
    pb_sales_amount = pb_items['sales_price'].sum()
    pb_item_percentage = (pb_item_count / total_items) * 100 if total_items != 0 else 0
    pb_sales_percentage = (pb_sales_amount / total_sales_amount) * 100 if total_sales_amount != 0 else 0
    print(f"Number of PB items: {pb_item_count}")
    print(f"Total sales amount of PB items: {pb_sales_amount}")
    print(f"Percentage of PB items: {pb_item_percentage:.2f}%")
    print(f"Percentage of sales amount of PB items: {pb_sales_percentage:.2f}%")
    
    # Non-PB items
    non_pb_item_count = len(non_pb_items)
    non_pb_sales_amount = non_pb_items['sales_price'].sum()
    non_pb_item_percentage = (non_pb_item_count / total_items) * 100 if total_items != 0 else 0
    non_pb_sales_percentage = (non_pb_sales_amount / total_sales_amount) * 100 if total_sales_amount != 0 else 0
    print(f"Number of Non-PB items: {non_pb_item_count}")
    print(f"Total sales amount of Non-PB items: {non_pb_sales_amount}")
    print(f"Percentage of Non-PB items: {non_pb_item_percentage:.2f}%")
    print(f"Percentage of sales amount of Non-PB items: {non_pb_sales_percentage:.2f}%")
    print()

Non_sale_Peak:
Number of PB items: 845
Total sales amount of PB items: 34668.8
Percentage of PB items: 4.30%
Percentage of sales amount of PB items: 3.81%
Number of Non-PB items: 18797
Total sales amount of Non-PB items: 876183.5999999999
Percentage of Non-PB items: 95.70%
Percentage of sales amount of Non-PB items: 96.19%

Non_sale_Non_peak:
Number of PB items: 299
Total sales amount of PB items: 14210.9
Percentage of PB items: 4.19%
Percentage of sales amount of PB items: 5.22%
Number of Non-PB items: 6836
Total sales amount of Non-PB items: 257885.49999999997
Percentage of Non-PB items: 95.81%
Percentage of sales amount of Non-PB items: 94.78%

Sale_Peak:
Number of PB items: 2825
Total sales amount of PB items: 187416.4
Percentage of PB items: 6.17%
Percentage of sales amount of PB items: 6.68%
Number of Non-PB items: 42954
Total sales amount of Non-PB items: 2617477.0
Percentage of Non-PB items: 93.83%
Percentage of sales amount of Non-PB items: 93.32%

Sale_Non_peak:
Number of PB 

In [77]:
# PB and NonPB for members
for df, name in zip(data_frames, ['Non_sale_Peak', 'Non_sale_Non_peak', 'Sale_Peak', 'Sale_Non_peak']):
    
    member_transactions = df[df['member_id'] != 'nonmember']
    
    # PB items for members
    pb_member_items = member_transactions[member_transactions['product_name'].str.contains('TV|TVHC', case=False, na=False)]
    non_pb_member_items = member_transactions[~member_transactions['product_name'].str.contains('TV|TVHC', case=False, na=False)]
    
    total_member_items = len(member_transactions)
    total_member_sales_amount = member_transactions['sales_price'].sum()

    print(f"{name} (Members):")
    
    # PB items for members
    pb_member_item_count = len(pb_member_items)
    pb_member_sales_amount = pb_member_items['sales_price'].sum()
    pb_member_item_percentage = (pb_member_item_count / total_member_items) * 100 if total_member_items != 0 else 0
    pb_member_sales_percentage = (pb_member_sales_amount / total_member_sales_amount) * 100 if total_member_sales_amount != 0 else 0
    print(f"Number of PB items (Members): {pb_member_item_count}")
    print(f"Total sales amount of PB items (Members): {pb_member_sales_amount}")
    print(f"Percentage of PB items (Members): {pb_member_item_percentage:.2f}%")
    print(f"Percentage of sales amount of PB items (Members): {pb_member_sales_percentage:.2f}%")
    
    # Non-PB items for members
    non_pb_member_item_count = len(non_pb_member_items)
    non_pb_member_sales_amount = non_pb_member_items['sales_price'].sum()
    non_pb_member_item_percentage = (non_pb_member_item_count / total_member_items) * 100 if total_member_items != 0 else 0
    non_pb_member_sales_percentage = (non_pb_member_sales_amount / total_member_sales_amount) * 100 if total_member_sales_amount != 0 else 0
    print(f"Number of Non-PB items (Members): {non_pb_member_item_count}")
    print(f"Total sales amount of Non-PB items (Members): {non_pb_member_sales_amount}")
    print(f"Percentage of Non-PB items (Members): {non_pb_member_item_percentage:.2f}%")
    print(f"Percentage of sales amount of Non-PB items (Members): {non_pb_member_sales_percentage:.2f}%")
    print()

Non_sale_Peak (Members):
Number of PB items (Members): 212
Total sales amount of PB items (Members): 9128.399999999998
Percentage of PB items (Members): 3.47%
Percentage of sales amount of PB items (Members): 2.82%
Number of Non-PB items (Members): 5894
Total sales amount of Non-PB items (Members): 314989.2
Percentage of Non-PB items (Members): 96.53%
Percentage of sales amount of Non-PB items (Members): 97.18%

Non_sale_Non_peak (Members):
Number of PB items (Members): 104
Total sales amount of PB items (Members): 5028.899999999999
Percentage of PB items (Members): 3.92%
Percentage of sales amount of PB items (Members): 4.77%
Number of Non-PB items (Members): 2549
Total sales amount of Non-PB items (Members): 100313.9
Percentage of Non-PB items (Members): 96.08%
Percentage of sales amount of Non-PB items (Members): 95.23%

Sale_Peak (Members):
Number of PB items (Members): 1402
Total sales amount of PB items (Members): 101573.5
Percentage of PB items (Members): 6.08%
Percentage of sal

In [78]:
# PB and NonPB for nonmembers
for df, name in zip(data_frames, ['Non_sale_Peak', 'Non_sale_Non_peak', 'Sale_Peak', 'Sale_Non_peak']):
    
    nonmember_transactions = df[df['member_id'] == 'nonmember']
    
    # PB items for members
    pb_nonmember_items = nonmember_transactions[nonmember_transactions['product_name'].str.contains('TV|TVHC', case=False, na=False)]
    non_pb_nonmember_items = nonmember_transactions[~nonmember_transactions['product_name'].str.contains('TV|TVHC', case=False, na=False)]
    
    total_nonmember_items = len(nonmember_transactions)
    total_nonmember_sales_amount = nonmember_transactions['sales_price'].sum()

    print(f"{name} (Nonembers):")
    
    # PB items for nonmembers
    pb_nonmember_item_count = len(pb_nonmember_items)
    pb_nonmember_sales_amount = pb_nonmember_items['sales_price'].sum()
    pb_nonmember_item_percentage = (pb_nonmember_item_count / total_nonmember_items) * 100 if total_nonmember_items != 0 else 0
    pb_nonmember_sales_percentage = (pb_nonmember_sales_amount / total_nonmember_sales_amount) * 100 if total_nonmember_sales_amount != 0 else 0
    print(f"Number of PB items (Nonmembers): {pb_nonmember_item_count}")
    print(f"Total sales amount of PB items (Nonmembers): {pb_nonmember_sales_amount}")
    print(f"Percentage of PB items (Nonmembers): {pb_nonmember_item_percentage:.2f}%")
    print(f"Percentage of sales amount of PB items (Nonmembers): {pb_nonmember_sales_percentage:.2f}%")
    
    # Non-PB items for nonmembers
    non_pb_nonmember_item_count = len(non_pb_nonmember_items)
    non_pb_nonmember_sales_amount = non_pb_nonmember_items['sales_price'].sum()
    non_pb_nonmember_item_percentage = (non_pb_nonmember_item_count / total_nonmember_items) * 100 if total_nonmember_items != 0 else 0
    non_pb_nonmember_sales_percentage = (non_pb_nonmember_sales_amount / total_nonmember_sales_amount) * 100 if total_nonmember_sales_amount != 0 else 0
    print(f"Number of Non-PB items (Nonmembers): {non_pb_nonmember_item_count}")
    print(f"Total sales amount of Non-PB items (Nonmembers): {non_pb_nonmember_sales_amount}")
    print(f"Percentage of Non-PB items (Nonmembers): {non_pb_nonmember_item_percentage:.2f}%")
    print(f"Percentage of sales amount of Non-PB items (Nonmembers): {non_pb_nonmember_sales_percentage:.2f}%")
    print()

Non_sale_Peak (Nonembers):
Number of PB items (Nonmembers): 633
Total sales amount of PB items (Nonmembers): 25540.399999999998
Percentage of PB items (Nonmembers): 4.68%
Percentage of sales amount of PB items (Nonmembers): 4.35%
Number of Non-PB items (Nonmembers): 12903
Total sales amount of Non-PB items (Nonmembers): 561194.4
Percentage of Non-PB items (Nonmembers): 95.32%
Percentage of sales amount of Non-PB items (Nonmembers): 95.65%

Non_sale_Non_peak (Nonembers):
Number of PB items (Nonmembers): 195
Total sales amount of PB items (Nonmembers): 9181.999999999998
Percentage of PB items (Nonmembers): 4.35%
Percentage of sales amount of PB items (Nonmembers): 5.51%
Number of Non-PB items (Nonmembers): 4287
Total sales amount of Non-PB items (Nonmembers): 157571.59999999998
Percentage of Non-PB items (Nonmembers): 95.65%
Percentage of sales amount of Non-PB items (Nonmembers): 94.49%

Sale_Peak (Nonembers):
Number of PB items (Nonmembers): 1423
Total sales amount of PB items (Nonmemb

第四个数据集1911_1011（SaleDay & NonPeakHour）存在两点异常：第一，会员交易量占比极低，其余三个数据集会员交易数量所占比为30%~50%，这个数据集会员交易量只有2.6%；第二，该数据集记录的会员交易中，所购买的产品全部为PB产品，不符实际。

**总体销售情况：**

1.促销活动可以显著提高所有产品的销售量，尤其是在高峰时段（注：但利润未知）；

2.不管是高峰时段还是非高峰时段，促销日中PB产品的销售量占比有显著增加，说明促销日是促进PB产品销售的有效方式。


**会员购买行为：**

1.非会员对于PB产品的购买占比更大，表明PB产品可能不是吸引会员的主要因素。


**策略建议：**

1.加强促销活动

*1.1 高峰时段和促销日安排更多促销活动，尤其是针对PB产品的促销活动（有可能现在的促销已经加强了对PB产品的促销，因此导致了“总体销售情况”中第二点的发生，在这种情况下此建议可能不再适用）；*

*1.2 细化对会员和非会员的促销活动，提供专属优惠积分奖励（一会搜索一下现在的优惠政策），进一步提升PB产品的销售。*

2.会员营销

*2.1 针对会员推出定期的促销活动和特别优惠，以增强会员的忠诚度和购买频率；*

*2.2 利用会员数据进行精准营销，推送个性化的商品推荐和促销信息，提高会员的参与度和购买意愿。*